examples/ui/drag_drop.rs

```diff
@@ -0,0 +1,374 @@
+                                    margin: UiRect {
+                                        left: Val::Auto,
+                                        right: Val::Auto,
+                                        ..default()
+                                    },
+                                    ..default()
+                                }),
+                            )
+                            .insert(Interaction::default())
+                            .insert(FocusPolicy::default())
+                            .insert(Dragable);
+                    }
+                });
+            // Drop area
+
+            parent
+                .spawn_bundle(NodeBundle {
+                    style: Style {
+                        size: Size::new(Val::Px(500.0), Val::Px(300.0)),
+                        position: UiRect::new(
+                            Val::Px(50.0),
+                            Val::Undefined,
+                            Val::Undefined,
+                            Val::Undefined,
+                        ),
+                        flex_direction: FlexDirection::Column,
+                        ..default()
+                    },
+                    color: Color::rgba(0.5, 0.1, 0.1, 1.).into(),
+                    ..default()
+                })
+                .insert(Interaction::default())
+                .insert(FocusPolicy::default())
+                .insert(DropContainer::default())
+                .with_children(|parent| {
+                    parent.spawn_bundle(TextBundle::from_section(
+                        "Drop into free space!".to_string(),
+                        TextStyle {
+                            font: asset_server.load("fonts/FiraMono-Medium.ttf"),
+                            font_size: 20.,
+                            color: Color::GRAY,
+                        },
+                    ));
+                });
+        });
+}
+
+fn mouse_hover(
+    mut event_mouse_enter: EventWriter<MouseEnterEvent>,
+    mut query_start_hover: Query<(Entity, &Interaction), (Changed<Interaction>, Without<MouseIn>)>,
+    mut event_mouse_exit: EventWriter<MouseExitEvent>,
+    mut query_stop_hover: Query<(Entity, &Interaction), (Changed<Interaction>, With<MouseIn>)>,
+) {
+    for (entity, interaction) in query_start_hover.iter_mut() {
+        match *interaction {
+            Interaction::Clicked | Interaction::None => {}
+            Interaction::Hovered => {
+                event_mouse_enter.send(MouseEnterEvent(entity));
+            }
+        }
+    }
+
+    for (entity, interaction) in query_stop_hover.iter_mut() {
+        match *interaction {
+            Interaction::Clicked | Interaction::Hovered => {}
+            Interaction::None => {
+                event_mouse_exit.send(MouseExitEvent(entity));
+            }
+        }
+    }
+}
+
+fn mouse_enter(mut mouse_enter_event: EventReader<MouseEnterEvent>, mut commands: Commands) {
```

Comment:
This indirection feels awkward to me; why not just insert the components directly in the systems that generate these events?

Reply:
Well, I wanted to provided hooking points for the drag and drop but its not useful for this example impl

Reply:
Ah yep, I agree with this analysis. For an example, I would simplify. Then when we build out a real solution we can migrate back to an event-powered design.